roles/public_facing/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+---
+## Common tasks
+
+# Most of our public-facing hosts are running Ubuntu.
+# use_ufw defaults to false but is overridden in inventory host_vars
+- include: ufw.yml
+  when: use_ufw == true
+  tags:
+    - always
+
+- include: fail2ban.yml
+  tags:
+    - always
+  when: use_fail2ban == true
+
+- name: Disable password authentication
+  lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: "^PasswordAuthentication"
+    line: "PasswordAuthentication no"
+    state: present
```

Comment:
You may need to restart/reload openssh here, or it would let server in insecure state.